buildSrc/src/main/kotlin/io/spine/internal/gradle/javac/Javac.kt

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2021, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.internal.gradle.javac
+
+import org.gradle.api.GradleException
+import org.gradle.api.JavaVersion
+import org.gradle.api.tasks.compile.JavaCompile
+import org.gradle.process.CommandLineArgumentProvider
+
+/**
+ * The knowledge that is required to set up `javac`.
+ */
+internal object JavacConfig {
+    const val SOURCE_FILES_ENCODING = "UTF-8"
+    val EXPECTED_JAVA_VERSION = JavaVersion.VERSION_1_8
+    val ARGUMENTS = CommandLineArgumentProvider {
+        listOf(
+
+            // Protobuf Compiler generates the code, which uses the deprecated `PARSER` field.
+            // See issue: https://github.com/SpineEventEngine/config/issues/173
+            // "-Werror",
+
+            "-Xlint:unchecked",
+            "-Xlint:deprecation",
+        )
+    }
+}
+
+/**
+ * Sets up `javac` by applying [JavacConfig].
+ *
+ * Here's an example of how to use it:
+ *
+ * ```
+ * tasks {
+ *     withType<JavaCompile> {
+ *         configureJavac()
+ *     }
+ * }
+ * ```
+ */
+fun JavaCompile.configureJavac() {
```

Comment:
And this goes first in this file. 

Beside the documentation it already has, it should document a few more points.

* Explain the idea of running with JDK 8.
* Tell what it does to `javac`.

The idea is that when someone navigates to this file, the first thing she sees is the entry point. Which is documented in a way that the gist of what's going to happen is already there. 

Now, it's not quite so: the private type goes first, the entry point goes last, and — for some reason — ping-pongs the reader back the the first type.

This comment applies to `ErrorProne` implementation as well, as it's pretty much the same as this one.